rclcpp/src/rclcpp/clock.cpp

```diff
@@ -30,29 +32,36 @@ JumpHandler::JumpHandler(
   notice_threshold(threshold)
 {}
 
-Clock::Clock(rcl_clock_type_t clock_type)
+static
+void
+rcl_clock_deleter(rcl_clock_t * rcl_clock)
 {
-  allocator_ = rcl_get_default_allocator();
-  auto ret = rcl_clock_init(clock_type, &rcl_clock_, &allocator_);
+  auto ret = rcl_clock_fini(rcl_clock);
   if (ret != RCL_RET_OK) {
-    exceptions::throw_from_rcl_error(ret, "could not get current time stamp");
+    RCUTILS_LOG_ERROR("Failed to fini rcl clock.");
   }
+  delete rcl_clock;
 }
 
-Clock::~Clock()
+Clock::Clock(rcl_clock_type_t clock_type)
 {
-  auto ret = rcl_clock_fini(&rcl_clock_);
+  allocator_ = rcl_get_default_allocator();
+  auto clock_unique_ptr = std::make_unique<rcl_clock_t>();
+  rcl_clock_ = std::shared_ptr<rcl_clock_t>(clock_unique_ptr.release(), rcl_clock_deleter);
```

Comment:
@ivanpauno fyi: I see what you're doing here, but it's not better than `new rcl_clock_t()` right there on line 50..

Reply:
Yeah, the temporary `unique_ptr` is only useful if some exception may occur between allocating and initializing the clock. Unless I miss something, I agree with hidmic that `new rcl_clock_t` will work just as well.